### ISSUE TITLE
Update to 2.14 fails

### ISSUE THREAD

**Issue by epi** [aBothe/Mono-D#636]
MonoDevelop 5.10, Ubuntu 14.04 LTS x86-64
Runtime:
	Mono 4.2.1 (Stable 4.2.1.102/6dd2d0d Thu Nov 12 09:52:44 UTC 2015) (64-bit)
	GTK+ 2.24.23 (Ambiance theme)

When updating from 2.13.5 to 2.14 in MonoDevelop's Add-in manager > Updates, an update failure is reported:

Exception occurred: Type 'MonoDevelop.D.Projects.Dub.DefinitionFormats.DubFileFormat' not found in add-in 'MonoDevelop.D,2.14'

System.InvalidOperationException: Type 'MonoDevelop.D.Projects.Dub.DefinitionFormats.DubFileFormat' not found in add-in 'MonoDevelop.D,2.14'
  at Mono.Addins.RuntimeAddin.GetType (System.String typeName, Boolean throwIfNotFound) <0x416d2d70 + 0x00387> in <filename unknown>:0 
  at Mono.Addins.TypeExtensionNode.get_Type () <0x416fec10 + 0x0004f> in <filename unknown>:0 
  at Mono.Addins.TypeExtensionNode.CreateInstance () <0x416febe0 + 0x0000f> in <filename unknown>:0 
  at Mono.Addins.InstanceExtensionNode.GetInstance () <0x416feb60 + 0x00023> in <filename unknown>:0 
  at Mono.Addins.ExtensionNodeEventArgs.get_ExtensionObject () <0x4174b6b0 + 0x0004f> in <filename unknown>:0 
  at MonoDevelop.Projects.ProjectService.OnFormatExtensionChanged (System.Object s, Mono.Addins.ExtensionNodeEventArgs args) <0x4174b3b0 + 0x0017f> in <filename unknown>:0 
  at Mono.Addins.ExtensionNode.OnChildNodeRemoved (Mono.Addins.ExtensionNode node) <0x40d54f00 + 0x0006b> in <filename unknown>:0 
  at Mono.Addins.ExtensionNode.NotifyChildChanged () <0x40d54a60 + 0x00100> in <filename unknown>:0 
  at Mono.Addins.TreeNode.NotifyChildrenChanged () <0x40d64e60 + 0x00023> in <filename unknown>:0 
  at Mono.Addins.TreeNode.Remove () <0x40d64de0 + 0x00063> in <filename unknown>:0 
  at Mono.Addins.ExtensionContext.RemoveAddinExtensions (System.String id) <0x40d641f0 + 0x001c7> in <filename unknown>:0 
  at Mono.Addins.AddinEngine.UnloadAddin (System.String id) <0x40d63f00 + 0x0003f> in <filename unknown>:0 
  at Mono.Addins.Database.AddinDatabase.DisableAddin (System.String domain, System.String id, Boolean exactVersionMatch) <0x40d630c0 + 0x0061f> in <filename unknown>:0 
  at Mono.Addins.Addin.set_Enabled (Boolean value) <0x40d84300 + 0x000d3> in <filename unknown>:0 
  at Mono.Addins.Gui.AddinManagerDialog.OnEnableDisableClicked (System.Object sender, System.EventArgs e) <0x40d84160 + 0x000cb> in <filename unknown>:0

**Comment by epi** [comment]
After a few restarts of MonoDevelop (without installing or removing anything), I was able to enable the updated D add-in.

**Comment by aBothe** [comment]
So after all, it's working?

**Comment by epi** [comment]
Yes, it's working now. I don't know what could be the reason that it refused to enable the add-in before.

**Comment by jpf91** [comment]
I had exactly the same problem on Archlinux, MonoDevelop 5.10. I got that Exception on the update and couldn't enable the addon anymore. Even deleting the addon, manually removing it from `.local` etc didn't help. In the end I deleted `.config/MonoDevelop .config/MonoDevelop-5.0 .cache/MonoDevelop-5.0/ .local/share/MonoDevelop-5.0`, reinstalled the addon and reconfigured MonoDevelop. Now everything is working as expected.
